packages/flutter_tools/lib/src/version.dart

```diff
@@ -722,7 +722,12 @@ class GitTagVersion {
 
   static GitTagVersion determine(ProcessUtils processUtils, {String workingDirectory, bool fetchTags = false}) {
     if (fetchTags) {
-      _runGit('git fetch $_flutterGit --tags', processUtils, workingDirectory);
+      final String channel = _runGit('git rev-parse --abbrev-ref HEAD', processUtils, workingDirectory);
```

Comment:
nit. can you use the `channel` getter from here https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/version.dart#L111

Reply:
I can't because instantiating that type calls this method

Reply:
doh! :)